src/framework/mpe/mpetypes.h

```diff
@@ -228,55 +233,57 @@ inline bool isMultiNoteArticulation(const ArticulationType type)
     return !isSingleNoteArticulation(type);
 }
 
+using dynamic_level_t = percentage_t;
+constexpr dynamic_level_t MAX_DYNAMIC_LEVEL = HUNDRED_PERCENTS;
+constexpr dynamic_level_t MIN_DYNAMIC_LEVEL = 0;
+constexpr dynamic_level_t DYNAMIC_LEVEL_STEP = 5 * SINGLE_PERCENT;
+
 enum class DynamicType {
     Undefined = -1,
-    pppppppp = 0,
-    ppppppp = 1,
-    pppppp = 2,
-    ppppp = 3,
-    pppp = 4,
-    ppp = 5,
-    pp = 6,
-    p = 7,
-    mp = 8,
-    Natural = 9,
-    mf = 10,
-    f = 11,
-    ff = 12,
-    fff = 13,
-    ffff = 14,
-    fffff = 15,
-    ffffff = 16,
-    fffffff = 17,
-    ffffffff = 18,
+    ppppppppp = MIN_DYNAMIC_LEVEL,
+    pppppppp = 5 * SINGLE_PERCENT,
+    ppppppp = 10 * SINGLE_PERCENT,
+    pppppp = 15 * SINGLE_PERCENT,
+    ppppp = 20 * SINGLE_PERCENT,
+    pppp = 25 * SINGLE_PERCENT,
+    ppp = 30 * SINGLE_PERCENT,
+    pp = 35 * SINGLE_PERCENT,
+    p = 40 * SINGLE_PERCENT,
+    mp = 45 * SINGLE_PERCENT,
+    Natural = 50 * SINGLE_PERCENT,
+    mf = 55 * SINGLE_PERCENT,
+    f = 60 * SINGLE_PERCENT,
+    ff = 65 * SINGLE_PERCENT,
+    fff = 70 * SINGLE_PERCENT,
+    ffff = 75 * SINGLE_PERCENT,
+    fffff = 80 * SINGLE_PERCENT,
+    ffffff = 85 * SINGLE_PERCENT,
+    fffffff = 90 * SINGLE_PERCENT,
+    ffffffff = 95 * SINGLE_PERCENT,
+    fffffffff = MAX_DYNAMIC_LEVEL,
     Last
 };
 
-using dynamic_level_t = percentage_t;
-constexpr dynamic_level_t MAX_DYNAMIC_LEVEL = 1.f;
-constexpr dynamic_level_t MIN_DYNAMIC_LEVEL = 0.f;
-constexpr dynamic_level_t DYNAMIC_LEVEL_STEP = MAX_DYNAMIC_LEVEL / (static_cast<int>(DynamicType::Last) - 1);
-
 inline DynamicType approximateDynamicType(const dynamic_level_t dynamicLevel)
 {
     if (dynamicLevel < MIN_DYNAMIC_LEVEL) {
-        return DynamicType::pppppppp;
+        return DynamicType::ppppppppp;
     }
 
     if (dynamicLevel > MAX_DYNAMIC_LEVEL) {
-        return DynamicType::ffffffff;
+        return DynamicType::fffffffff;
     }
 
-    return static_cast<DynamicType>(std::roundf(dynamicLevel / DYNAMIC_LEVEL_STEP));
+    return static_cast<DynamicType>(dynamicLevel);
 }
 
-inline dynamic_level_t dynamicLevelFromType(const DynamicType type)
+constexpr inline dynamic_level_t dynamicLevelFromType(const DynamicType type)
```

Comment:
It may make sense to remove `DynamicType` from `MPE`, since this is an entity from the notation (visual)

Reply:
I'd leave it here like other essential types (pitch classes, articulation types, etc.) It's not really visual types, they're all about the different characteristics of the sound